src/cerebellum.py

```diff
@@ -0,0 +1,114 @@
+"""Pathfinding Module"""
+import pdb
+from queue import Queue
+from typing import List, Optional
+from src.snake import Snake
+from src.board import Board
+from src.coordinate import Coordinate
+
+
+class Cerebellum:
+    """Find paths to given coordinates."""
+    def __init__(self, brain, selected_algorithm: Optional[str]) -> None:
+        self.selected_algorithm: str = selected_algorithm if selected_algorithm is not None else 'breadth_first'
+        self.brain = brain
+        self.me = self.brain.me
+        self.board: Board = self.brain.board
+
+    def get_path(self, coordinate: Optional[Coordinate] = None, coordinates: Optional[List[Coordinate]] = None) -> List[Coordinate]:
+        """Find path to given coordinate/coordinates based on pathfinder's selected algorithm."""
+
+        if coordinates is None:
+            if coordinate is None:
+                return []
+            coordinates = [coordinate]
+
+        path: List[Coordinate] = []
+
+        if self.selected_algorithm == 'breadth_first':
+            path = self.__get_breadth_first_path(coordinates)
+
+#        elif self.selected_algorithm == 'djikstra\'s':
+#            path = self.__get_djikstras_path(coordinates)
+
+#        elif self.selected_algorithm == 'a*':
+#            path = self.__get_a_star_path(coordinates)
+
+        return path
+
+    def __get_breadth_first_path(self, coordinates: List[Coordinate]) -> List[Coordinate]:
+        """Find path using breadth-first algorithm."""
+
+        #pick location from grid (self.me.head)
+        #expand that location by looking at neighbours
+        #any unvisited neighbours are added to "unexplored" and "visited" set
+
+        for coord in coordinates:
+            print(coord)
+
+        start = self.me.head
+
+        unexplored: Queue = Queue()
+        unexplored.put(start)
+        came_from: dict = {}
+        came_from[str(start)] = None
+
+        collision_coordinates = [coordinate for snake in self.board.snakes for coordinate in snake.coordinates]
+
+        #remove this snake's tail from array of "invalid moves"
+        tail = self.brain.me.coordinates[-1]
+        if tail in collision_coordinates:
+            collision_coordinates.remove(tail)
+
+
+        matched = False
+        while not unexplored.empty():
+            current = unexplored.get()
+
+            if any(coord.x == current.x and coord.y == current.y for coord in coordinates):
```

Comment:
I think I'd prefer passing in a single coordinate, and then calling the search multiple times, but maybe that's not a great solution for breadth-first. If you prefer it this way, no problem.

Reply:
I'm not sure I see the benefit -- but it could certainly be worth trying.
However, since this is working, and competition's tomorrow, maybe we compromise on a trello task to explore that option?